cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/ResponseFormatUtils.scala

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.services.openai
+
+object ResponseFormatUtils {
+  private val SchemaHints = Set(
+    "properties",
+    "required",
+    "additionalProperties",
+    "$schema",
+    "items"
+  )
+  private val JsonTypeKeywords = Set(
+    "object",
+    "array",
+    "string",
+    "number",
+    "integer",
+    "boolean",
+    "null"
+  )
+
+  def normalize(value: Map[String, Any]): Map[String, Any] = {
+    requireNonNull(value)
+    val tOpt = rfType(value)
+    val isTextOrJson = tOpt.exists(t => t == "text" || t == "json_object")
+    val isJsonSchemaT = tOpt.contains("json_schema")
+    val shouldFlatten = tOpt match {
+      case Some(t) => JsonTypeKeywords.contains(t) || looksLikeInnerSchema(value)
+      case None    => isInnerSchemaCandidate(value)
+    }
+
+    if (isTextOrJson) {
+      Map("type" -> tOpt.get)
+    } else if (isJsonSchemaT) {
+      normalizeJsonSchema(value)
+    } else if (shouldFlatten) {
+      flattenToFlatJsonSchema(value)
+    } else if (tOpt.isEmpty) {
+      requireBareJsonSchemaShape(value)
+      Map("type" -> "json_schema") ++ value
+    } else {
+      val other = tOpt.get
+      throw new IllegalArgumentException(
+        "Unsupported response_format type: '" + other + "'. Allowed: 'text', 'json_object', 'json_schema'.")
+    }
+  }
+
+  private def normalizeJsonSchema(value: Map[String, Any]): Map[String, Any] = {
+    val hasNested = value.contains("json_schema")
+    val hasFlat = value.contains("name") && value.contains("schema")
+    if (!hasNested && !hasFlat) {
+      throw new IllegalArgumentException(
+        "json_schema requires nested 'json_schema' or top-level 'name' and 'schema'.")
+    }
+    if (hasFlat) {
```

Comment:
Should we error/alert if the user erroneously specifies both json_schema + name/schema?